app/controllers/articles_controller.rb

```diff
@@ -35,4 +36,16 @@ def set_categories
       active_category category.parent_id if category.child?
     end
   end
+
+  def set_show_newsletter_signup
+    exclusions = YAML.load(
+      File.read(File.expand_path("#{Rails.root}/config/newsletter_article_exclusion_list.yml", __FILE__))
```

Comment:
Personally I would create a new class that would return the array of articles and not bother with the yml file. That object can change when you want it to read from another source (i.e. something a admin/non-dev can change)